core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/version/VersionSearchRequest.java

```diff
@@ -112,7 +116,37 @@ protected Expression prepareQuery(RepositoryContext context) {
 
 		return query.build();
 	}
-
+	
+	protected final void addResourcesFilter(ExpressionBuilder queryBuilder, RepositoryContext context) {
+		User user = context.service(User.class);
+		
+		if (containsKey(OptionKey.RESOURCE)) {
+			Collection<String> resources = getCollection(OptionKey.RESOURCE, String.class);
+			Collection<String> resourceIds = resources.stream()
+					.map(resource -> resource.contains("/") ? new ResourceURI(resource).getResourceId() : resource)
+					.toList();
+			
+			if (user.isAdministrator()) {
+				queryBuilder.filter(VersionDocument.Expressions.resourceIds(resourceIds));
+			} else {
+				final AuthorizationService authz = context.optionalService(AuthorizationService.class).orElse(AuthorizationService.DEFAULT);
+				Set<String> accessibleResources = authz.getAccessibleResources(context, context.service(User.class));
+				
+				resourceIds.removeIf(resourceId -> !accessibleResources.contains(resourceId));
+				queryBuilder.filter(VersionDocument.Expressions.resourceIds(resourceIds));
```

Comment:
If `resourceIds` becomes empty after authorization check, the search request can return quickly by throwing a NoResultException.